src/likelihoods/gaussian.jl

```diff
@@ -20,3 +20,23 @@ GaussianLikelihood() = GaussianLikelihood(1e-6)
 (l::GaussianLikelihood)(f::Real) = Normal(f, sqrt(l.σ²))
 
 (l::GaussianLikelihood)(fs::AbstractVector{<:Real}) = MvNormal(fs, sqrt(l.σ²))
+
+"""
+    HeteroscedasticGaussianLikelihood(σ²)
+
+Heteroscedastic Gaussian likelihood. This is to be used if we assume that the 
+uncertainity associated with the data follows a Gaussian distribution where the 
+variance varies according to the input.
+
+```math
+    p(y|[f, σ]) = Normal(y | f, σ)
```

Comment:
Could you rewrite this as `p(y|f,g) = Normal(y | f, exp(g))`
That leaves it clear that sigma is another gp